### ISSUE TITLE
"Add mods" section isn't self populating?

### ISSUE THREAD

**Issue by randomname42** [Youda008/DoomRunner#169]
It's hard to tell if this is a bug or intentional, but this certainly feels like a bug. After adding a folder for iwads and a folder for maps, those lists self populated. After adding a mod folder (with mods already in it), a mod list did not self populate. 

That is confusingly inconsistent, if it's intentional. At first I thought the mods section was broken because of that. I _can_ manually add mods one by one to each preset, so apparently it is functional in that respect. But it's not very convenient to manually populate the mods list one mod at a time, and have to redo that for each preset.

There are screenshots in the readme page with a lot of of mods listed, which would also make it seem that the mods list is supposed to self populate.

If it's not a bug that the mods list isn't self populating, please, please add that function. After finding out about this launcher I've found it to mostly be great, and I've been setting up presets for a lot of maps I want to play. But it's quickly becoming a real pain to have to manually add every mod to the list one by one, and to have to do that every time for every preset.

It's the only seriously inconvenient thing I can think of after using DoomRunner for a couple days now.

Possibly useful info, if mods not self populating is a bug:

I'm on a Steam Deck using flatpak. I added the correct folder permissions to Flatseal for DoomRunner. Because I can manually add mods to the mods list, this bug doesn't seem related to folder/flatpak permissions.

When I first went through initial setup, I only set up iwad and map folders, as I wanted to make sure permissions were working correctly before messing around too much. Those both self populated, so then I went to initial setup again and added the mods folder. 

Could that somehow have caused mods to not auto populate (because I didn't add a mods folder the first time)? If so, is there anything I can do to fix it?

**Comment by Youda008** [comment]
It's not a bug. It is intentional and it has been working like that from the beginning.
The main reason is that if it populated automatically, if would be sorted alphabetically and you wouldn't be able to change the load order, which is crucial.
Also, there are people (including me), who don't have a separate folder for every mod and instead have all mod files in a single folder and then cherry-pick which one go in which preset.
You don't have to add the files one by one, you can add multiple files at once, both from the Select File dialog that opens when you click the Add button, or by using drag&drop from a Windows File Explorer window.

**Comment by rye-74** [comment]
i agree with youda on this, i have all my mods and wads in a single folder as well.

**Comment by EtiamNullam** [comment]
It's confusing as in "Intial setup" you can define "Directory with Mods". Not sure what is the purpose of it. It's also next to "Directory with Maps" which is self-populating.

![image](https://github.com/user-attachments/assets/d4e9e44d-a7da-40f3-93ed-3e0a700e70d8)

**Comment by Youda008** [comment]
This Directory with Mods only serves as a starting directory that opens when you click the Add button.
I get it. This design isn't flawless, but most people seem pretty happy with it, and changing it now would introduce much more issues than it would solve.

**Comment by Youda008** [comment]
I have removed the mod directory field, it was no longer used for anything and it will remove this confusion.
